### PR TITLE
add shared ig contacts to private pages

### PR DESCRIPTION
**Pull request by ruslandoga**
This probably shouldn't be merged but rather ran once.

### CODE REVIEW

File: dev/dev.ex [hunk 1]
```diff
@@ -46,4 +46,340 @@ defmodule Dev do
       end
     end
   end
+
+  @ig_contacts [
```
Comment (ruslandoga) [inline dev/dev.ex:50]: you'd need to regenerate this list and edit it manually since when you run it, it'd likely be different

File: dev/dev.ex [hunk 2]
```diff
@@ -46,4 +46,340 @@ defmodule Dev do
...
+
+      if has_contacts?(story, ig) do
+        acc
+      else
+        [%{user_id: user_id, story: add_private_page(story, ig)} | acc]
+      end
+    end)
+  end
+
+  def insert_new_stories(stories \\ add_private_pages()) do
```
Comment (ruslandoga) [inline dev/dev.ex:344]: once the list is ready, `Dev.insert_new_stories` would replace stories in db

File: dev/dev.ex [hunk 3]
```diff
@@ -46,4 +46,101 @@ defmodule Dev do
...
     end
   end
+
+  @ig_contacts [
+    %{ig: ["@durov"], user_id: "0000017c-0000-0000-0000-ac1100040000"}
+  ]
+
+  import Ecto.Query
+
+  # use this function to get a list like above
```
Comment (akhrail1996) [inline dev/dev.ex:56]: IO.inspect(contacts, limit: :ifinity)

**Review by akhrail1996 — APPROVED** [review]

File: dev/dev.ex (hunk 3 above)
Comment (ruslandoga) [reply dev/dev.ex:56]: sorry, it's :infinity

File: dev/dev.ex [hunk 4]
```diff
@@ -46,4 +46,101 @@ defmodule Dev do
...
+    |> Enum.reject(fn %{ig: ig} -> is_nil(ig) end)
+    |> Enum.map(fn %{ig: igs} = i -> %{i | ig: filter_valid_igs(igs)} end)
+    |> Enum.reject(fn %{ig: ig} -> ig == [] end)
+  end
+
+  defp filter_valid_igs(igs) do
+    Enum.filter(igs, fn ig ->
+      Regex.match?(
+        ~r/^[A-Za-z0-9_](?:(?:[A-Za-z0-9_]|(?:\.(?!\.))){0,99}(?:[A-Za-z0-9_]))$/,
+        ig |> String.replace(["@"], "") |> String.trim()
```
Comment (ruslandoga) [inline dev/dev.ex:81]: this needs to be run before filtering otherwise we get invalid username like `@durov  `

**Comment by akhrail1996** [comment]
done

**Pull request closed without merging**